### PR TITLE
Cleanups related to legacy nid support

### LINKED ISSUE #14393: Resolve or remove TODO 3.0 related to legacy NID support in crypto/evp
There are these following TODOs related to legacy NID support in evp_enc.c:
```
/*
 * FIPS module note: since internal fetches will be entirely
 * provider based, we know that none of its code depends on legacy
 * NIDs or any functionality that use them.
 */
#ifndef FIPS_MODULE
/* TODO(3.x) get rid of the need for legacy NIDs */
static void set_legacy_nid(const char *name, void *vlegacy_nid)
```
and
```
static void *evp_cipher_from_dispatch(const int name_id,
                                      const OSSL_DISPATCH *fns,
                                      OSSL_PROVIDER *prov)
{
...
#ifndef FIPS_MODULE
    /* TODO(3.x) get rid of the need for legacy NIDs */
    cipher->nid = NID_undef;
    if (!evp_names_do_all(prov, name_id, set_legacy_nid, &cipher->nid)
```
In digest.c there are:
```
*
 * FIPS module note: since internal fetches will be entirely
 * provider based, we know that none of its code depends on legacy
 * NIDs or any functionality that use them.
 */
#ifndef FIPS_MODULE
/* TODO(3.x) get rid of the need for legacy NIDs */
static void set_legacy_nid(const char *name, void *vlegacy_nid)
```
and
```
static void *evp_md_from_dispatch(int name_id,
                                  const OSSL_DISPATCH *fns,
                                  OSSL_PROVIDER *prov)
{
...
#ifndef FIPS_MODULE
    /* TODO(3.x) get rid of the need for legacy NIDs */
    md->type = NID_undef;
    if (!evp_names_do_all(prov, name_id, set_legacy_nid, &md->type)
```

Those need to be resolved before beta1.

### PR DESCRIPTION
**Pull request by t8m**
Fixes #14393

### CODE REVIEW

File: apps/crl.c [hunk 1]
```diff
@@ -347,8 +347,7 @@ int crl_main(int argc, char **argv)
                     BIO_printf(bio_err, "out of memory\n");
                     goto end;
                 }
-                BIO_printf(bio_out, "%s Fingerprint=",
-                           OBJ_nid2sn(EVP_MD_type(digest)));
+                BIO_printf(bio_out, "%s Fingerprint=", EVP_MD_name(digest));
```
Comment (levitte) [inline apps/crl.c:350]: I would actually make this use `digestname` instead.

File: apps/x509.c [hunk 2]
```diff
@@ -978,7 +978,7 @@ int x509_main(int argc, char **argv)
                 BIO_printf(bio_err, "Out of memory\n");
                 goto end;
             }
-            BIO_printf(out, "%s Fingerprint=", OBJ_nid2sn(EVP_MD_type(fdig)));
+            BIO_printf(out, "%s Fingerprint=", EVP_MD_name(fdig));
```
Comment (levitte) [inline apps/x509.c:981]: Same here, use `digestname` (and fall back to `"SHA1"` if it's NULL)

**Review by levitte — CHANGES_REQUESTED** [review]

File: apps/crl.c (hunk 1 above)
Comment (t8m) [reply apps/crl.c]: Fixed

File: apps/x509.c (hunk 2 above)
Comment (t8m) [reply apps/x509.c]: Done

**Review by levitte — APPROVED** [review]

**Comment by t8m** [comment]
@levitte  the fixup broke one of the tests - I'll probably revert it. Of course also the test could be adjusted but thinking about this I do not think it is worth it and some third party scripts might also depend on that "canonical" value being used there.

**Comment by t8m** [comment]
@levitte OK to drop the fixup?

**Comment by t8m** [comment]
@levitte ping for the re-approval.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master. Thank you for the reviews!

**Pull request closed without merging**